lua/nvim-treesitter/utils.lua

```diff
@@ -76,4 +77,20 @@ function M.setup_commands(mod, commands)
   end
 end
 
+--- Gets the smallest scope which contains @param node
+function M.smallest_containing_scope(node, bufnr)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+
+  local root = parsers.get_parser(bufnr):parse():root()
+  if not node then return root end
+
+  local scopes = locals.get_scopes(bufnr)
+  local current = node
+  while current ~= nil and not vim.tbl_contains(scopes, current) do
+    current = current:parent()
+  end
+
+  return current or root
+end
```

Comment:
couldn't this function be in the `utils` file ? It looks like it could be used in other places.

Reply:
If you mean `tsutils` then yes, it should be there ;-)

Reply:
Actually, it is in the `utils` file